tests/CMakeLists.txt

```diff
@@ -1,16 +1,13 @@
 find_package(Eigen3 REQUIRED)
-find_package(Catch2 REQUIRED)
+find_package(Catch2 3 REQUIRED)
 
 file(GLOB_RECURSE TEST_FILES "*.test.cpp")
 add_executable(autodiff-cpptests main.cpp ${TEST_FILES})
-target_link_libraries(autodiff-cpptests autodiff::autodiff Eigen3::Eigen Catch2::Catch2)
+target_link_libraries(autodiff-cpptests autodiff::autodiff Eigen3::Eigen Catch2::Catch2WithMain)
 set_target_properties(autodiff-cpptests PROPERTIES CXX_STANDARD 17 CXX_STANDARD_REQUIRED ON CXX_EXTENSIONS OFF)
 
-# Add target tests that performs all C++ and Python tests
-add_custom_target(tests
```

Comment:
We need the target `tests`, which is used in the CI (e.g., `ninja tests`). This code here removes it.